lib/floe/workflow/payload_template.rb

```diff
@@ -21,10 +21,11 @@ def interpolate_value_nested(value, context, inputs)
           value.map { |val| interpolate_value_nested(val, context, inputs) }
         when Hash
           value.to_h do |key, val|
-            val = interpolate_value_nested(val, context, inputs)
-            key = key.gsub(/\.\$$/, "") if key.end_with?(".$")
-
-            [key, val]
+            if key.end_with?(".$")
+              [key.gsub(/\.\$$/, ""), interpolate_value_nested(val, context, inputs)]
```

Comment:
Better method for perf:

```suggestion
              [key.chomp(".$"), interpolate_value_nested(val, context, inputs)]
```